src/hooks/DynamicValuesHooks.js

```diff
@@ -115,3 +117,121 @@ export const useDynamicValues = (parameter, targetDatasetId) => {
     loadingDynamicValuesPlaceholder,
   };
 };
+
+export const useDynamicParameter = (parameterValue, parameter, targetDatasetId) => {
+  const { t } = useTranslation();
+  const findDatasetById = useFindDatasetById();
+  const organizationId = useOrganizationId();
+  const parametersValues = useCurrentScenarioParametersValues();
+
+  const isUnmounted = useRef(false);
+  useEffect(() => () => (isUnmounted.current = true), []);
+
+  // Possible value types for dynamicValue:
+  // - 'undefined' when dynamic values are not enabled in the parameter configuration
+  // - 'null' when fetching data, a placeholder with a spinner will be displayed instead of the input
+  // - a value when the query is successful, it will be displayed in the input
+  const [dynamicValue, setDynamicValue] = useState(undefined);
+  const [dynamicValueError, setDynamicValueError] = useState(null);
+  const defaultValue = parameter?.defaultValue ?? GENERIC_VAR_TYPES_DEFAULT_VALUES[parameter?.varType];
+
+  useEffect(() => {
+    if (isUnmounted.current) return;
+    const dynamicSourceConfig = parameter.options?.dynamicValues;
+    const scenarioParameterValue =
+      parametersValues.find((scenarioParameter) => scenarioParameter.parameterId === parameter.id) ?? null;
+    if (scenarioParameterValue !== null) {
+      setDynamicValueError(null);
+    }
+
+    const fetchDynamicValue = async () => {
+      if (parameterValue !== null) {
+        setDynamicValue(undefined);
+        return;
+      }
```

Comment:
I used `setDynamicValue(undefined)` to stop the display of loading spinner but now I'm not so sure that it's necessary, I will check :+1:

Reply:
I forgot to reply to this :woman_facepalming: I checked, indeed the switch to `undefined` is necessary to stop the loading spinner which is displayed when `dynamicValue` is `null` (its initial state before triggering the function) :slightly_smiling_face:

Reply:
My bad, my message was not clear, I was not questioning the call to `setDynamicValue` :sweat_smile: 
My question was only on the error part: do you think we could/should add `setDynamicValueError(null);` to reset the previous error (if any) when we receive a result from the query?

Reply:
Interesting question, I think it is not necessary because when the `useEffect` fires, the value of `dynamicValueError` is already set to `null` as it's its initial state?

Reply:
And I finally read your question the right way, sorry :sweat_smile:

Reply:
Ok, thanks for your answer :slightly_smiling_face: